WordPress/WordPressShareExtension/AppExtensionsService.swift

```diff
@@ -318,12 +327,18 @@ extension AppExtensionsService {
             return
         }
 
-        if let tags = tags {
-            remotePost.tags = tags.arrayOfTags()
-        }
+        switch type {
+        case .post:
+            remotePost.type = "post"
+            if let tags = tags {
+                remotePost.tags = tags.arrayOfTags()
+            }
 
-        if let remoteCategories = RemotePostCategory.remotePostCategoriesFromString(categories) {
-            remotePost.categories = remoteCategories
+            if let remoteCategories = RemotePostCategory.remotePostCategoriesFromString(categories) {
+                remotePost.categories = remoteCategories
+            }
+        case .page:
+            remotePost.type = "page"
```

Comment:
These 11 lines seem to be a perfect copy of the 11 lines in `saveAndUploadPost(...)`.  Any thoughts on possible improvements?

Reply:
Perhaps it makes more sense for these two methods to accept the `ShareData` as a parameter, and then a new `remotePost` initialiser could be made that accepts it, and we move the shared logic into there. Or, have the methods accept `RemotePost` and move that creation out of both. Either would be a good change, I think; @diegoreymendez would you have a preference?

```
saveAndUploadPost(shareData: ShareData, siteID: Int, [...])
uploadPostWithMedia(shareData: ShareData, siteID: Int, [...])
```
vs
```
saveAndUploadPost(remotePost: RemotePost, [...])
uploadPostWithMedia(remotePost: RemotePost, [...])
```

Reply:
Great observation!  I think both solutions are acceptable, and in fact I would encourage you to pick the one that best suites your preference.